lib/r10k/action/runner.rb

```diff
@@ -100,11 +100,22 @@ def settings_from_config(override_path)
       def add_credential_overrides(overrides)
         sshkey_path = @opts[:'private-key']
         token_path = @opts[:'oauth-token']
+        app_id = @opts[:'github-app-id']
+        app_private_key_path = @opts[:'github-app-key']
+        app_ttl = @opts[:'github-app-ttl']
 
         if sshkey_path && token_path
           raise R10K::Error, "Cannot specify both an SSH key and a token to use with this deploy."
         end
 
+        if sshkey_path && app_private_key_path || app_private_key_path && token_path || app_id && token_path || app_id && sshkey_path
+          raise R10K::Error, "Cannot specify both an SSH key and an SSL key or token to use with this deploy."
+        end
```

Comment:
Might be good to separate this into two checks? One for `sshkey_path` and `app_private_key_path` OR `app_id` and one for `token_path` and `app_private_key_path` OR `app_id`. Does that make sense?

Reply:
Thanks for the feedback @mwaggett. I have done manual testing, with a small mock-up environment using a control repo syncing three repositories. It works, and happy to share the output if interested. I will add the CHANGELOG entry, as well as your suggested changes. Thanks again